src/App.css

```diff
@@ -73,4 +75,18 @@ h1 {
 h3 {
   font-size: 1.3rem; 
   text-align: center;
+}
+
+.img-thumbnail {
+  object-fit: contain;
+  max-height: 500px;
+}
+
+p {
+  break-before: avoid-page;
```

Comment:
This one seems a bit too strong.  I would think that _between_ paragraphs would a reasonable place for a page break.